distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2500,6 +2506,78 @@ private void checkExpectations(CometDClient cometDClient, Integer numNotificatio
 
     }
 
+    /**
+     * Confirms that the "download resource to local cache" action's url can be
+     * found in the CometDClient messages and matches the expected url.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void findMetacardActionUrlUsingCometDClient() throws Exception {
+
+        String metacardId = ingestXmlWithProduct(String.format("%s.txt", testName.getMethodName()));
+
+        metacardsToDelete.add(metacardId);
+
+        final String RESPONSE_CHANNEL = "/0193d9e7f9ed4f8f8bd02103143c41d6";
+
+        cometDClient = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL,
+                QUERY_PUBLISH_CHANNEL,
+                RESPONSE_CHANNEL));
+
+        String src = "ddf.distribution";
+
+        makeAndPublishQueryData(RESPONSE_CHANNEL, src, metacardId);
+
+        expect("Waiting for response").within(20, SECONDS)
+                .until(() -> cometDClient.getMessages(RESPONSE_CHANNEL)
+                        .size() >= 1);
+
+        List<String> messages = cometDClient.getAllMessages();
+
+        LOGGER.debug("cometDClient messages size: {}", messages.size());
+        for (String msg : messages) {
+            LOGGER.debug("cometDClient message: {}", msg);
+        }
+
+        //The action title connected to the url that will be retrieved.
+        String title = "Download resource to local cache";
+
+        String url = String.format("%s?source=%s&metacard=%s", RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl(), src, metacardId);
+
+        //Finds the wanted url by searching for the metacardId and then finding the action by its title.
+        String desiredUrl =
+                cometDClient.findActionUrlBySearchCriterion(metacardId, title);
+
+        assertThat(desiredUrl, is(url));
+
+    }
+
```

Comment:
This method should be in the cometD client.  We should have a method on the cometD client (serachByMetacardId) that takes a metacardId and a response channel.  This way we could use it in other tests and not have to copy this private method.